runner/local/queueing_runner.go

```diff
@@ -0,0 +1,228 @@
+package local
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/scootdev/scoot/runner"
+	"golang.org/x/net/context"
+)
+
+// This runner manages multiple (possibly concurrent) run and status requests.
+// One request is run at a time, additional requests are queued till the current request is finished running.
+
+// QueueingRunner is initialized with the runner to use to run the requests, the maximum number of requests to
+// manage on the queue, and the 'callback' channel the runner will use to tell queueing runner that it has
+// completed the last request.
+//
+// When a request is received it is immediately added to the queue.  As the runner becomes available, if
+// if there is are requests on the queue, the top request on the queue is sent to the runner
+//
+// If the max queue length is exceed, QueueingRunner rejects any new request till length of the queue is below
+// the max.
+//
+// QueueingRunner assigns a unique id (run id) to each new request.  The Run() function returns this id and a
+// status indicating if the request has been successfully queued or if the request has been denied (due to queue
+// length exceeded or some other error).
+
+// The status of a request is obtained with the run id.
+
+const QueueFullMsg = "No resources available. Please try later."
+const UnspportedFeatureMsg = "Not implemented.  This feature is not available for QueueingRunner"
+
+// The request queue entries are commandAndId structs
+type commandAndId struct {
```

Comment:
My mistake, QueueRunner needs to generate its own runIds so it can return a valid run handle without actually running the command.